### PR TITLE
ARROW-5725: [Crossbow] Port conda recipes to azure pipelines 

### PR DESCRIPTION
**Pull request by kszucs**
- [x] artifact uploading
- [x] osx build
- [x] win build (using appveyor, because of https://github.com/conda-forge/conda-forge.github.io/issues/703)
- [x] linux build
- [x] package gandiva

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
One of the azure build has lost connection, but I cannot restart it on the UI.

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

**Comment by kszucs** [comment]
The builds are passing, but the artifacts are not uploaded correctly.

**Comment by kszucs** [comment]
@ursabot crossbow package -g conda

File: dev/tasks/docker-tests/azure.linux.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - script: |
+      git clone --no-checkout {{ arrow.remote }} arrow
+      git -C arrow fetch -t {{ arrow.remote }} {{ arrow.branch }}
+      git -C arrow checkout FETCH_HEAD
+      git -C arrow submodule update --init --recursive
+    displayName: Clone arrow
+
+  - script: |
+      pushd arrow
+      {%- for command in commands %}
```
Comment (pitrou) [inline dev/tasks/docker-tests/azure.linux.yml]: Hmm... and where are the commands defined?

**Review by pitrou — COMMENTED** [review]
Not sure I have anything constructive to say about this, sorry. Some questions though...

File: dev/tasks/conda-recipes/.ci_support/linux_python2.7.yaml [hunk 2]
```diff
@@ -1,40 +1,24 @@
 boost_cpp:
-- 1.68.0
+- 1.70.0
```
[the remark this answers is not in the document]
Comment (kszucs) [reply dev/tasks/conda-recipes/.ci_support/linux_python2.7.yaml]: By conda-smithy, we need to port after the migrations.

File: dev/tasks/conda-recipes/appveyor.win.yml [hunk 3]
```diff
@@ -1,20 +1,3 @@
-# Licensed to the Apache Software Foundation (ASF) under one
```
[the remark this answers is not in the document]
Comment (kszucs) [reply dev/tasks/conda-recipes/appveyor.win.yml]: Because these files are mostly ported from the [py]arrow[-cpp] feedstocks. There is a note in the LICENSE.txt that files under `dev/tasks/conda-recipes` have another license. Also added to the rat exclude list.

File: dev/tasks/conda-recipes/arrow-cpp/build.sh [hunk 4]
```diff
@@ -1,29 +1,18 @@
...
 
 cd cpp
 mkdir build-dir
 cd build-dir
 
+EXTRA_CMAKE_ARGS=""
+
+# Include g++'s system headers
+if [ "$(uname)" == "Linux" ]; then
+  SYSTEM_INCLUDES=$(echo | ${CXX} -E -Wp,-v -xc++ - 2>&1 | grep '^ ' | awk '{print "-isystem;" substr($1, 1)}' | tr '\n' ';')
```
[the remark this answers is not in the document]
Comment (kszucs) [reply dev/tasks/conda-recipes/arrow-cpp/build.sh]: I'm open to any suggestions.

File: dev/tasks/conda-recipes/azure.linux.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    CONFIG: {{ config }}
+    ARROW_VERSION: {{ arrow.no_rc_version }}
+    DOCKER_IMAGE: condaforge/linux-anvil-comp7
+    UPLOAD_PACKAGES: False
+
+  steps:
+  # configure qemu binfmt-misc running.  This allows us to run docker containers
+  # embedded qemu-static
+  - script: |
+      docker run --rm --privileged multiarch/qemu-user-static:register --reset --credential yes
```
[the remark this answers is not in the document]
Comment (kszucs) [reply dev/tasks/conda-recipes/azure.linux.yml]: Ported from the upstream feedstocks (it is generated but once written by someone).

File: dev/tasks/docker-tests/azure.linux.yml (hunk 1 above)
Comment (kszucs) [reply dev/tasks/docker-tests/azure.linux.yml]: In the [tests.yml](https://github.com/apache/arrow/blob/master/dev/tasks/tests.yml#L103)

**Comment by kszucs** [comment]
@ursabot crossbow package conda-win-vs2015-py36 conda-win-vs2015-py37

**Comment by kszucs** [comment]
@pitrou the artifacts are properly uploaded to github releases now

File: dev/tasks/conda-recipes/arrow-cpp/build.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (pitrou) [reply dev/tasks/conda-recipes/arrow-cpp/build.sh]: Unfortunately I have none, sorry.

**Comment by pitrou** [comment]
Side note, would it be possible for ursabot to post links to artifacts in its report above? Finding files in the github releases is a bit cumbersome.

**Comment by kszucs** [comment]
It is more complicated then the report above. 

1. `python crossbow.py submit ...` creates the branches and triggers the builds on various CIs
2. `python crossbow.py status ...` polls the github api about these statuses, checks and release assets

The ursabot crossbow command is not a service like buildbot, thus it is not notified about changes with hooks, and the `ursabot crossbow` command is currently only a proxy to `python crossbow.py submit ...`, executed via buildbot. 

On the long term I could port the functionality of crossbow into ursabot, but it requires the development of a series of `meta` services, a meta-scheduler, a meta-builder, a meta-worker. The scheduler would manipulate the crossbow repository directly (preferably via github api instead of native git bindings), the triggered meta builders would follow the logs of other services via their api and we'd be notified about crossbow's status and check updates by github hooks, then we could properly report back all of the results (including the artifacts).

TLDR it is possible, but requires more time.

**Comment by pitrou** [comment]
I don't mean post the links when they are available, but upfront (so they would give a 404 until the build is finished). It depends whether the URLs are deterministic or not...

**Comment by kszucs** [comment]
This is how the badges are working. For the conda packages the assets are not deterministic: https://github.com/apache/arrow/pull/4649/files#diff-0abe5ae3589de012350559065a4194a6R155

**Review by pitrou — APPROVED** [review]
+1. No objection from me.
